.github/workflows/ci.yml

```diff
@@ -1,32 +1,50 @@
-name: Unit Tests 
-
-# Controls when the action will run. Triggers the workflow on push or pull request
-# events but only for the master branch
-on: 
-  pull_request:
+name: Unit Tests
+on:
   push:
-  workflow_dispatch:
-
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+permissions:
+  actions: write
+  contents: read
 env:
+  DATADEPS_ALWAYS_ACCEPT: 1
   JIVE_PROCS: 1
-
 jobs:
-  build:
-    runs-on: ubuntu-20.04
-
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.6'
+          - '1'
+          - 'nightly'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
     steps:
-      
-      - uses: actions/checkout@v2
-      - uses: julia-actions/setup-julia@latest
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@v2
         with:
-          version: 1.6
-
-      # Runs a single command using the runners shell
-      - name: Unit Tests
-        run: |
-          julia --project -e 'using Pkg; Pkg.instantiate(); Pkg.build();'
-          julia --project --check-bounds=yes --depwarn=yes -e 'import Pkg; Pkg.test(; coverage=true)'
-      
-      - name: Codecov Upload
-        run: |
-          julia -e 'using Pkg; Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder());'
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v2
+        with:
+          file: lcov.info
+  docs:
+    name: Documentation
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: julia-actions/julia-buildpkg@latest
+      - uses: julia-actions/julia-docdeploy@latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
```

Comment:
In the past we wanted to trigger doc only builds wihtout running tests but I guess merging them should be fine.

Reply:
Should be okay to merge them.